lib/vmpooler/providers/vsphere.rb

```diff
@@ -1103,6 +1103,9 @@ def migrate_vm(pool_name, vm_name)
             begin
               connection = ensured_vsphere_connection(pool_object)
               vm_hash = get_vm_details(pool_name, vm_name, connection)
+
+              raise StandardError, 'failed to get vm details. vm is unreachable or no longer exists' if vm_hash.nil?
```

Comment:
Suggest adding details about where it was trying to get vm details. Kindof like this but may need something fancier than `.to_s` for the arguments.

```ruby
raise StandardError, "failed to get VM details from pool: #{pool_name}, vm_name: #{vm_name}, connection: #{connection}" ...
```

Reply:
The error message actually gets logged in the vmpooler code with a message that includes the vm name and pool. I think the connection might be an object, but am not certain on that.

Reply:
https://github.com/puppetlabs/vmpooler/blob/3327cc22269fd5b0e28fd1499cfaf383a5fdfdd9/lib/vmpooler/pool_manager.rb#L864